simulation/tests/test_costs.py

```diff
@@ -174,3 +175,63 @@ def test_air_source_heat_pumps_get_cheaper_across_2022(self):
                 assert quote == future_quote
 
             quote = future_quote
+
+    @pytest.mark.parametrize("boiler", set(BOILERS))
+    def test_boiler_upgrade_scheme_grant_is_always_zero_for_boilers(self, boiler):
```

Comment:
I suggest (with the other change below) that this test just tests that it is `always_zero_for_boilers_within_grant_window`. Then, the test below will test that it is zero for boilers outside the grant window.